### ISSUE TITLE
BaseAsset validation cause error registering a module which imported from npm package

### ISSUE THREAD

**Issue by nazarhussain** [LiskHQ/lisk-sdk#6477]
### Expected behavior

Registering a module which is published as npm package should not cause any problem. 

```js
import {MyModule} from 'my-module-package';

app.registerModule(MyModule);
```

### Actual behavior

With the above scenario if we register a module which contains some asset right now cause following error. 

> Custom module contains asset which does not extend `BaseAsset` class.

### Steps to reproduce
1. Create an npm package with `npm init` 
2. Add `lisk-sdk` as dependency in that package 
3. Develop your module `MyModule` with some assets in it and export from npm packge
4. Now bootstrap an application with `lisk init` 
5. Add that module with `npm i -s my-module-package`
6. Register the module with the application 

### Which version(s) does this affect? (Environment, OS, etc...)
5.x.x

**Comment by nazarhussain** [comment]
We need to identity other use cases of `BasePlugin` and `BaseModule` and come up with a better dependency validation for the non-compliant or mis-matching SDK versions.

**Comment by Hupka** [comment]
Heyey, I want add something: the issue is only observed when importing a _locally_ available npm package through `npm install ../path/to/module/folder` which creates an entry in the `package.json` with the format:

```JSON
{
  "dependencies": {
    "myModule": "file:../path/to/module/folder"
  }
}
```

The use case is the development of a standalone Lisk Module. During development one wants to also test the Lisk Module in a "host application" (= Lisk Blockchain Application) which resides in some other folder locally. Since both, module and host application, contain a `node_modules` folder one strong guess is that the respectively Lisk-related installed npm packages create conflicts when attempting to start the host application through `./bin/run start`.

**Comment by shuse2** [comment]
Resolved with #6554
